src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/ProxyService.scala

```diff
@@ -76,19 +74,23 @@ class ProxyService(proxyConfig: ProxyConfig,
       if (!hasViewPermission) {
         throw ClusterNotFoundException(googleProject, clusterName)
       } else if (!hasRequiredPermission) {
-        throw AuthorizationError(userInfo.userEmail)
+        throw AuthorizationError(Option(userInfo.userEmail))
       } else {
         ()
       }
     }
   }
 
-  def proxyLocalize(userInfo: UserInfo, googleProject: GoogleProject, clusterName: ClusterName, request: HttpRequest, token: HttpCookiePair): Future[HttpResponse] = {
+  def proxyLocalize(userInfo: UserInfo, googleProject: GoogleProject, clusterName: ClusterName, request: HttpRequest): Future[HttpResponse] = {
     authCheck(userInfo, googleProject, clusterName, SyncDataToCluster).flatMap { _ =>
-      proxyInternal(userInfo, googleProject, clusterName, request, token)
+      proxyInternal(userInfo, googleProject, clusterName, request)
     }
   }
 
+  def invalidateAccessToken(token: String): Future[Unit] = {
+    Future(cachedAuth.invalidate(token))
+  }
```

Comment:
Two questions.

1. This doesn't blacklist the token, right? It just removes it from the cache. So if another request comes in with the same token in a cookie, we'll go right back and fetch its creds from Google? If that's the case, what does this achieve?

2. Following from this, I think this means if Leo is bounced while a user has a notebook session open, their connection should resume fine? We'll repopulate the cache once from Google and then continue serving up their content as normal? That's nice :)

Reply:
1. Correct, it just removes the token from cache, so any subsequent request to Leo will cause us to re-fetch the creds from Google. It's for the case when the user signs out of the UI (invalidating their token in Google) while they have a notebook open. We want the notebook to stop working at that point too -- not continue working for 1 hour until the cache entry expires.

2. Yeah, as long as the token is valid in Google Leo should just repopulate the cache after it is restarted.

Reply:
Ah great, that makes sense. Thanks!